check/run-check.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+if [ ! -f typediff ]; then
+    wget https://github.com/haskell-compat/base-compat/releases/download/typediff-0.1.0/typediff
+    chmod +x typediff
+fi
+PATH=./:$PATH cabal test
```

Comment:
Ok, ic, here we download `typediff`.  Maybe ditch that script altogether and put it in the readme that a user has to

    (cd ../typediff/ && cabal install)

before running the tests?  At least I don't really like to run scripts that download binary executables and run them without any form of trust metric.